contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -2043,6 +2042,36 @@ class tsqlBuilder : public tsqlCommonMutator
 	{
 		graft(makeExecBodyBatch(ctx), peekContainer());
 	}
+
+	PLtsql_expr *rewrite_if_condition(TSqlParser::Search_conditionContext *ctx)
+	{
+		PLtsql_expr *expr = makeTsqlExpr(ctx, false);
+		PLtsql_expr_query_mutator mutator(expr, ctx);
+		add_rewritten_query_fragment_to_mutator(&mutator);
+		mutator.run();
```

Comment:
Usually we call `clear_rewritten_query_fragment()`. Is it okay not calling it? (Ref: https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/BABEL_3_X_DEV/contrib/babelfishpg_tsql/src/tsqlIface.cpp#L178-L211)

Reply:
yes, we should call it